active/0000-production-code-stripping.md

```diff
@@ -0,0 +1,56 @@
+- Start Date: 2016-04-06
+- RFC PR: (leave this empty)
+- ember-cli Issue: (leave this empty)
+
+# Summary
+
+A number of Ember framework function calls are no-ops in production. Ember CLI should strip these no-op function invocations from production builds by default.
+
+# Motivation
+
+Removing code that isn't required in production results in smaller and faster applications.
+
+# Detailed design
+
+The following framework function calls will be removed from ember-cli production builds by default:
+
+ * [`Ember.assert`](http://emberjs.com/api/#method_assert)
+ * [`Ember.debug`](http://emberjs.com/api/#method_debug)
+ * [`Ember.deprecate`](http://emberjs.com/api/#method_deprecate)
+ * [`Ember.info`](http://emberjs.com/api/#method_info)
+ * [`Ember.runInDebug`](http://emberjs.com/api/#method_runInDebug)
+ * [`Ember.warn`](http://emberjs.com/api/#method_warn)
+
+The API documentation will be updated where necessary to indicate that these function calls will be stripped from production.
+
+A Babel plugin will execute the removal of these function calls. It will either be:
+
+ * [babel-plugin-filter-imports](https://github.com/ember-cli/babel-plugin-filter-imports) with added support for defining module functions for removal ([spike here](https://github.com/GavinJoyce/babel-plugin-filter-imports/pull/1))
+ * a new babel plugin which works similarly to `babel-plugin-filter-imports`
```

Comment:
Thanks. This one doesn't seem to handle import statements, but it is useful for reference